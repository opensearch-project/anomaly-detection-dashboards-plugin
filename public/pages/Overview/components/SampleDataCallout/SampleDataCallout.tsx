/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { EuiCallOut, EuiLink } from '@elastic/eui';
import {
  APP_PATH,
  PLUGIN_NAME,
  BASE_DOCS_LINK,
} from '../../../../utils/constants';

export const SampleDataCallout = () => {
  return (
    <EuiCallOut
      title="Looking to get more familiar with anomaly detection?"
      color="primary"
      iconType="help"
    >
      <p>
        Read the{' '}
        <EuiLink href={`${BASE_DOCS_LINK}/ad`} target="_blank">
          documentation
        </EuiLink>{' '}
        or create a{' '}
        <EuiLink href={`${PLUGIN_NAME}#${APP_PATH.SAMPLE_DETECTORS}`}>
          sample detector
        </EuiLink>{' '}
        to get started.
      </p>
    </EuiCallOut>
  );
};
