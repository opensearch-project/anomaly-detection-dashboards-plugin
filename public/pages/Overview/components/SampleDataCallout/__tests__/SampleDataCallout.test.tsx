/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { SampleDataCallout } from '../SampleDataCallout';

describe('<SampleDataCallout /> spec', () => {
  describe('Data not loaded', () => {
    test('renders component', () => {
      const { container, getByText } = render(<SampleDataCallout />);
      expect(container.firstChild).toMatchSnapshot();
      getByText('Looking to get more familiar with anomaly detection?');
    });
  });
});
