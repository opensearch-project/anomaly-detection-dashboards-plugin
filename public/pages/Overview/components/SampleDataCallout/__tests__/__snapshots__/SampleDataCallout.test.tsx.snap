// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleDataCallout /> spec Data not loaded renders component 1`] = `
<div
  class="euiCallOut euiCallOut--primary"
>
  <div
    class="euiCallOutHeader"
  >
    <svg
      aria-hidden="true"
      class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
      focusable="false"
      height="16"
      role="img"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
      />
    </svg>
    <span
      class="euiCallOutHeader__title"
    >
      Looking to get more familiar with anomaly detection?
    </span>
  </div>
  <div
    class="euiText euiText--small"
  >
    <div
      class="euiTextColor euiTextColor--default"
    >
      <p>
        Read the
         
        <a
          class="euiLink euiLink--primary"
          href="https://opensearch.org/docs/monitoring-plugins/ad"
          rel="noopener noreferrer"
          target="_blank"
        >
          documentation
          <svg
            aria-hidden="true"
            aria-label="External link"
            class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
            />
          </svg>
          <span
            class="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
         
        or create a
         
        <a
          class="euiLink euiLink--primary"
          href="anomaly-detection-dashboards#/overview"
          rel="noreferrer"
        >
          sample detector
        </a>
         
        to get started.
      </p>
    </div>
  </div>
</div>
`;
